tests/db_functions/test_datetime.py

```diff
@@ -254,6 +260,47 @@ def test_extract_day_func(self):
         )
         self.assertEqual(DTModel.objects.filter(start_datetime__day=ExtractDay('start_datetime')).count(), 2)
 
+    def test_extract_week_func(self):
+        start_datetime = microsecond_support(datetime(2015, 6, 15, 14, 30, 50, 321))
+        end_datetime = microsecond_support(datetime(2016, 6, 15, 14, 10, 50, 123))
+        if settings.USE_TZ:
+            start_datetime = timezone.make_aware(start_datetime, is_dst=False)
+            end_datetime = timezone.make_aware(end_datetime, is_dst=False)
+        self.create_model(start_datetime, end_datetime)
+        self.create_model(end_datetime, start_datetime)
+        self.assertQuerysetEqual(
+            DTModel.objects.annotate(extracted=ExtractWeek('start_datetime')).order_by('start_datetime'),
+            [(start_datetime, 25), (end_datetime, 24)],
+            lambda m: (m.start_datetime, m.extracted)
+        )
+        self.assertQuerysetEqual(
+            DTModel.objects.annotate(extracted=ExtractWeek('start_date')).order_by('start_datetime'),
+            [(start_datetime, 25), (end_datetime, 24)],
+            lambda m: (m.start_datetime, m.extracted)
+        )
+        # both dates are from the same week.
+        self.assertEqual(DTModel.objects.filter(start_datetime__week=ExtractWeek('start_datetime')).count(), 2)
+
+        week_52_day_2014 = microsecond_support(datetime(2014, 12, 27, 13, 0))  # Sunday
```

Comment:
As long as the second part isn't dependent on the first, I'd add a separate test method such as:
`def test_extract_week_func_boundaries`